File: leetcode/c++/34. Find First and Last Position of Element in Sorted Array.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> search(int nums[],int target,int len){
  int left = 0;
  int right = len-1;
  vector<int> res(2,-1);
  bool has= false;
  while (left <= right) {
            int mid = left + (int)((right - left) / 2);
            if (nums[mid] < target) {
                left++;
            } else if (nums[mid] == target) {
                has = true;
                right--;
            } else {
                right--;
            }
        }
  if (has) {
            res[0] = left;
  }
  left = 0;
        right = len - 1;
        while (left <= right) {
            int mid = left + (int)((right - left) / 2);
            if (nums[mid] <= target) {
                left++;
            } else {
                right--;
            }
        }
        if (has) {
            res[1] = right;
  }
  return res;
}

int main(){
  int nums[]{5,7,7,8,8,10};
  int len = sizeof(nums)/sizeof(nums[0]);
  vector<int> v = search(nums,8,len);
  for(auto i:v){
    cout << i << endl;
  }
  return 0;
}

File: leetcode/kotlin/34. Find First and Last Position of Element in Sorted Array.kt
internal object Main {
    private fun searchRange(nums: IntArray, target: Int): IntArray {
        var left = 0
        var right = nums.size - 1
        val res = IntArray(2)
        res[0] = -1
        var has = false
        while (left <= right) {
            val mid = left + (right - left) / 2
            if (nums[mid] < target) {
                left++
            } else if (nums[mid] == target) {
                has = true
                right--
            } else {
                right--
            }
        }
        if (has) {
            res[0] = left
        }
        left = 0
        right = nums.size - 1
        while (left <= right) {
            val mid = left + (right - left) / 2
            if (nums[mid] <= target) {
                left++
            } else {
                right--
            }
        }
        if (has) {
            res[1] = right
        }
        return res
    }

    @JvmStatic
    fun main(s: Array<String>) {
        println(searchRange(intArrayOf(5, 7, 7, 8, 8, 10), 8).contentToString())
    }
}